python/hopsworks_common/core/type_systems.py

```diff
@@ -184,7 +218,15 @@ def cast_pandas_column_to_offline_type(
     ):
         return feature_column.apply(
             lambda x: (ast.literal_eval(x) if isinstance(x, str) else x)
-            if (x is not None and x != "")
+            if (
+                x is not None
+                and (
+                    not pd.isnull(x)
+                    if not isinstance(x, (list, dict, np.ndarray))
+                    else True
+                )
+                and (x != "" if not isinstance(x, (list, dict, np.ndarray)) else True)
+            )
```

Comment:
wouldn't this be the same?
```suggestion
            if (
                x is not None
                and (
                    isinstance(x, (list, dict, np.ndarray))
                    or (not pd.isnull(x) and x != "")
                )
            )
```

Reply:
I agree I changed it.